model-optimizer/extensions/middle/ApplyPermutations.py

```diff
@@ -141,7 +141,10 @@ def permute_input_data(graph: Graph):
 
                 if permutation_data_node.has_and_set('permutation') and \
                         not is_input_data_in_correct_layout(node, in_port) and \
-                        len(port_to_check.data.get_shape()) >= 4:
+                        (len(port_to_check.data.get_shape()) >= 4 or \
+                            (node['type'] == 'StridedSlice' and node.in_port(in_port).data.get_shape()[0] >= 4)):
```

Comment:
`node['type']` -> `node.soft_get('type')`

Reply:
Don't understand why you check that the first dimension of input is >=4

Reply:
changed this logic, these lines are not present any more. About necessity to permute even if input has rank <= 4 but slice_rank (first dimension of input) is >= 4 discussed personally and have written in comments here
1. `model-optimizer/mo/graph/perm_inputs.py:246`
2. `model-optimizer/mo/graph/perm_inputs.py:131`
3. `model-optimizer/extensions/middle/StridedSliceNormalizer.py:33`